flink-java/src/main/java/org/apache/flink/api/java/utils/ParameterTool.java

```diff
@@ -192,6 +192,7 @@ public static ParameterTool fromSystemProperties() {
 	 * @throws IOException If arguments cannot be parsed by {@link GenericOptionsParser}
 	 * @see GenericOptionsParser
 	 */
+	@Deprecated
```

Comment:
Adding the dependencies org.apache.hadoop in flink-java only for one method of this class looks not very good. The method fromGenericOptionsParser will be moved with another Hadoop dependent methods into flink-hadoop-compatibility. This method isn't used inside flink. If anybody uses this in his code, he should rewrite before migrating to next major version of flink.
You can look to the full version flink-java without Hadoop here https://github.com/kenmy/flink/tree/FLINK-4048